### PR TITLE
Run golangci-lint in every Go project in repository (and fix all reported errors)

### PR DESCRIPTION
**Pull request by mrnugget**
I noticed that `golangci-lint` only runs over the packages belonging to `github.com/sourcegraph/sourcegraph`. It doesn't run over `lib`, `dev/sg`, etc.

This PR changes the check script we use in CI to run golangci-lint over every project in our repository with a `go.mod` (which is what we do for `go test` in CI too).

That's the first commit. The second commit fixes all of the reported errors.

And since a lot of the reported errors were about dead code (that I deleted) I urge you to take a look, ask yourself whether you'll miss the dead code and **if yes: hop on this branch here, bring the code back from the grave & make it undead so that golangci-lint doesn't complain anymore**

### CODE REVIEW

**Review by eseliger — APPROVED** [review]

File: lib/output/capabilities.go [hunk 1]
```diff
@@ -21,7 +21,8 @@ func detectCapabilities() (capabilities, error) {
 	w, h := 80, 25
 	var err error
 	if atty {
-		size, err := term.GetWinsize(os.Stdout.Fd())
+		var size *term.Winsize
+		size, err = term.GetWinsize(os.Stdout.Fd())
```
Comment (LawnGnome) [inline lib/output/capabilities.go:25]: Oof to me.

**Review by LawnGnome — APPROVED** [review]
LGTM, although I don't have the context for the code intel parts of this PR to know for sure that they're safe.

File: dev/sg/main.go [hunk 2]
```diff
@@ -189,8 +190,9 @@ See the "checks:" in the configuration file.`,
 	funkyLogoFlagSet = flag.NewFlagSet("sg logo", flag.ExitOnError)
 	funkLogoCommand  = &ffcli.Command{
 		Name:       "logo",
-		ShortUsage: "sg logo",
+		ShortUsage: "sg logo [classic]",
 		ShortHelp:  "Print the sg logo",
+		LongHelp:   "Prints the sg logo in different colors. When the 'classic' argument is passed it prints the logo that's used in install.sh",
```
Comment (efritz) [inline dev/sg/main.go:195]: _I can't believe this was lint_

File: lib/codeintel/lsif/conversion/correlate.go [hunk 3]
```diff
@@ -84,6 +84,10 @@ func CorrelateLocalGit(ctx context.Context, dumpPath, projectRoot string) (*prec
 	getChildrenFunc := pathexistence.LocalGitGetChildrenFunc(gitRoot)
 
 	relRoot, err := filepath.Rel(gitRoot, absoluteProjectRoot)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Error getting relative path of %q and %q", gitRoot, absoluteProjectRoot)
```
Comment (efritz) [inline lib/codeintel/lsif/conversion/correlate.go:88]:
```suggestion
		return nil, errors.Wrapf(err, "failed to get relative path of %q and %q", gitRoot, absoluteProjectRoot)
```

File: lib/codeintel/tools/lsif-repl/main.go [hunk 4]
```diff
@@ -194,37 +185,3 @@ func queryBundle(bundle *precise.GroupedBundleDataMaps, path string, line, chara
 	}
 	return nil
 }
-
-// finds all documents which have definition edges pointing into the argument list
-func documentsReferencing(bundle *precise.GroupedBundleDataMaps, paths []string) (_ []string, err error) {
```
Comment (efritz) [inline lib/codeintel/tools/lsif-repl/main.go:199]: I wonder if this is used in one of Roux's unmerged branches. If so we can just remember that this function is here.

**Review by efritz — APPROVED** [review]

**Review by unknwon — APPROVED** [review]

File: dev/sg/main.go (hunk 2 above)
Comment (mrnugget) [reply dev/sg/main.go:195]: Welcome to the future!

File: lib/output/capabilities.go (hunk 1 above)
Comment (mrnugget) [reply lib/output/capabilities.go:25]: In this PR we're all Oof.

**Pull request merged**